lib/rspec/mocks/any_instance/stub_chain.rb

```diff
@@ -30,7 +30,7 @@ def invocation_order
             :with => [nil],
             :and_return => [:with, nil],
             :and_raise => [:with, nil],
-            :and_yield => [:with, nil],
+            :and_yield => [:with, :and_yield, nil],
```

Comment:
I'm not sure this is the way to fix this, other combinations would cause the same error as before, we need to return `self` from all the method calls that support further chaining I think.